### ISSUE TITLE
chdir: No such file or directory

### ISSUE THREAD

**Issue by bsideup** [cloudfoundry-incubator/lattice#66]
```
FROM dockerfile/java:oracle-java8

ADD app.jar /app/app.jar

CMD java -Djava.awt.headless=true -server -jar /app/app.jar

EXPOSE 8080
```

this image will fail with `chdir: No such file or directory` error without `-w /app/` option.

Workaround: specify `-w /app/` option.

**Comment by onsi** [comment]
I looks like [`dockerfile/java:oracle-java8` defines `/data` to be the `WORKDIR`](https://github.com/dockerfile/java/blob/master/oracle-java8/Dockerfile)

You could add `WORKDIR /app` to your Dockerfile -- then you won't need `-w`

**Comment by bsideup** [comment]
@onsi this Dockerfile works great inside Docker :)

**Comment by onsi** [comment]
can you send me the logs you get out of `ltc create`

**Comment by bsideup** [comment]
```
No port specified, using exposed ports from the image metadata.
	Exposed Ports: 8080
No working directory specified, using working directory from the image metadata...
Working directory is:
/data
Monitoring the app on port 8080...
No start command specified, using start command from the image metadata...
Start command is:
/bin/sh -c java -Djava.awt.headless=true -server -jar /app/app.jar
Creating App: app2
.30 Mar 22:53 [APP|0] Creating container
.....30 Mar 22:53 [APP|0] Successfully created container
30 Mar 22:53 [APP|0] chdir: No such file or directory
30 Mar 22:53 [APP|0] Exit status 255
30 Mar 22:53 [APP|0] Creating container
.....30 Mar 22:53 [APP|0] Successfully created container
30 Mar 22:53 [APP|0] chdir: No such file or directory
30 Mar 22:53 [APP|0] Exit status 255
30 Mar 22:53 [APP|0] Creating container
.....30 Mar 22:53 [APP|0] Successfully created container
30 Mar 22:53 [APP|0] chdir: No such file or directory
30 Mar 22:53 [APP|0] Exit status 255
............................................
app2 took too long to start.
```

And here is logs from `ltc dl`: https://gist.github.com/bsideup/daf6f26121dc01d01ef5

**Comment by onsi** [comment]
@bsideup - this one's definitely interesting ;)

I was hoping to take a look today but ran out of time.  Will try again tomorrow - is there a repo you can point me to (either github or docker) that will let me reproduce exactly this issue?

**Comment by bsideup** [comment]
@onsi since Dockefile is simple (see this issue details) I can easily setup repo for you little bit later. You can reproduce it by yourself as well I think

**Comment by onsi** [comment]
I think I've identified the root cause.  Docker automatically creates creates the working directory if it does not already exists.  Garden does not do this.  I'll open an issue in Garden-Linux.

**Comment by davidwadden** [comment]
This was addressed with [[#91550852](https://www.pivotaltracker.com/story/show/91550852)]; fixed in CF Garden [**v0.290.0**](https://github.com/cloudfoundry-incubator/garden-linux-release/releases/tag/v0.290.0), landed in CF Diego on [**0.1412.0**](https://github.com/cloudfoundry-incubator/diego-release/releases/tag/0.1412.0), which was upgraded to [**0.1424.1**](https://github.com/cloudfoundry-incubator/diego-release/releases/tag/0.1424.1) with Lattice [**v0.5.0**](https://github.com/cloudfoundry-incubator/lattice/releases/tag/v0.5.0).

@bsideup Can you retry with our latest version and let us know if it works?

**Comment by davidwadden** [comment]
Closing this issue.  Re-open if this issue resurfaces.

Thanks again!
